Maple2.Model/Enum/Portal.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace Maple2.Model.Enum;
+
+public enum PortalType : byte {
```

Comment:
some notes:
Type 4 and 11 are the only ones that set `ActionType`
![image](https://github.com/kOchirasu/Maple2/assets/7710340/f6799bc4-0c8d-4d19-a2ef-1656f841a009)

type `{4}` is `Portal_Type_A`
type `{3, 5, 6, 7, 10, 11, 16}` are `MS2RoomEnterPortal`

Reply:
That makes sense for 11 (don't know what 4 is). Considering Portal is a record, how would you recommend adjusting this to set the action type?

Reply:
isn't action type already set from xml?

Reply:
if you need to mutate, it's the same as readonly structs and you can use `with` syntax to copy with mutation.

```cs
Portal a;
Portal b = a with {ActionType = VALUE};
```

Reply:
Yes and no. Field portals set by the xblocks are already correctly set.
Portals from the hidden realms/bonus maps spawn objects are not found in the xblocks, hence why they're being set manually on FieldManager.SpawnEventPortal

But even then, those portals are Interact only (because they're portal type 3)